connect/connect.go

```diff
@@ -0,0 +1,292 @@
+package connect
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/kyma-incubator/hydroform/connect/types"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+func (c *KymaConnector) Connect(configurationUrl string) error {
+
+	_, err := c.populateCsrInfo(configurationUrl)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	_, err = c.populateClientCert()
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	if c.SecureClient == nil {
+		err = c.populateClient()
+		if err != nil {
+			return fmt.Errorf(err.Error())
+		}
+	}
+
+	_, err = c.populateInfo()
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	err = c.StorageInterface.WriteClientCert(c.Ca)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	return err
+}
+
+func (c *KymaConnector) RegisterService(serviceDescription Service) (serviceId string, err error) {
+	jsonBytes, err := json.Marshal(serviceDescription)
+	if err != nil {
+		return "", fmt.Errorf(err.Error())
+	}
+
+	if c.CsrInfo == nil || c.CsrInfo.API.MetadataUrl == "" {
+		return "", fmt.Errorf(err.Error())
+	}
+
+	resp, err := c.SecureClient.Post(c.CsrInfo.API.MetadataUrl, "application/json", bytes.NewBuffer(jsonBytes))
+	if err != nil {
+		return "", fmt.Errorf(err.Error())
```

Comment:
Why converting an error to a string to convert it to an error again? You can just return the error:
```suggestion
		return "", err
```

Ideally you might want to add some context to where the error happened, someting like:
```suggestion
		return "", errors.Wrap(err, "Error marshalling service description")
```